src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -0,0 +1,373 @@
+
+        Args:
+            pixel_values_list (`List[torch.Tensor]`):
+                List of images (pixel values) to be padded. Each image should be a tensor of shape (C, H, W).
+            return_tensors (`str` or [`~file_utils.TensorType`], *optional*):
+                If set, will return tensors instead of NumPy arrays. If set to `'pt'`, return PyTorch `torch.Tensor`
+                objects.
+
+        Returns:
+            [`BatchFeature`]: A [`BatchFeature`] with the following fields:
+
+            - **pixel_values** -- Pixel values to be fed to a model.
+            - **pixel_mask** -- Pixel mask to be fed to a model (when `pad_and_return_pixel_mask=True` or if
+              *"pixel_mask"* is in `self.model_input_names`).
+
+        """
+
+        max_size = self._max_by_axis([list(image.shape) for image in pixel_values_list])
+        c, h, w = max_size
+        pixel_values = []
+        pixel_mask = []
+        pixel_labels = []
+        class_labels = []
+
+        for idx, image in enumerate(pixel_values_list):
+            # create padded image
+            if should_pad:
+                padded_image = np.zeros((c, h, w), dtype=np.float32)
+                padded_image[: image.shape[0], : image.shape[1], : image.shape[2]] = np.copy(image)
+                image = padded_image
+            pixel_values.append(image)
+            # if we have a target, pad it
+            if annotations:
+                annotation = annotations[idx]
+                masks = annotation["masks"]
+                if should_pad:
+                    padded_masks = np.zeros((masks.shape[0], h, w), dtype=masks.dtype)
+                    padded_masks[:, : padded_masks.shape[1], : padded_masks.shape[2]] = np.copy(padded_masks)
+                    masks = padded_masks
+                pixel_labels.append(masks)
+                class_labels.append(annotation["labels"])
+            if should_pad:
+                # create pixel mask
+                mask = np.zeros((h, w), dtype=np.int64)
+                mask[: image.shape[1], : image.shape[2]] = True
+                pixel_mask.append(mask)
+
+        # return as BatchFeature
+        data = {"pixel_values": pixel_values, "pixel_mask": pixel_mask}
+
+        encoded_inputs = BatchFeature(data=data, tensor_type=return_tensors)
+        if annotations:
+            # BatchFeature doesn't support nested dicts, adding after it
+            tensor_type = return_tensors
+            if not isinstance(tensor_type, TensorType):
+                tensor_type = TensorType(tensor_type)
+            if not tensor_type == TensorType.PYTORCH:
+                raise ValueError("Only PyTorch is supported for the moment.")
+            else:
+                if not is_torch_available():
+                    raise ImportError("Unable to convert output to PyTorch tensors format, PyTorch is not installed.")
+
+                encoded_inputs["labels"] = {
```

Comment:
We shouldn't have a nested dict here for consistency with other models. The API should be:
```
features = FeatureExtracor(**inputs)
outputs = Model(**features)
```
e.g. the dict returned by the feature extractor should go in the model directly.

Models that expect several labels, have several arguments for each label (see DETR, or question answering models for instance). Here we should have `pixel_labels` and `class_labels`.

Reply:
Okay. One issue is API clarity. E.g. if you pass `pixel_labels` you have to pass  `class_labels` as well, and vice-versa, you can enforce only with conditional raising. Passing a single dict solved it. In DETR `labels` is used. @NielsRogge